specification/communication/Communication.JobRouter/models.tsp

```diff
@@ -0,0 +1,1527 @@
+
+@doc("Response payload from cancelling a job")
+model CancelJobResult {}
+
+alias CancelJobResultWithOkResponse = ResourceOkResponse<CancelJobResult>;
+
+@doc("Response payload from completing a job")
+model CompleteJobResult {};
+
+alias CompleteJobResultWithOkResponse = ResourceOkResponse<CompleteJobResult>;
+
+@doc("Response payload from declining a job")
+model DeclineJobOfferResult {};
+
+alias DeclineJobOfferResultWithOkResponse = ResourceOkResponse<DeclineJobOfferResult>;
+
+alias CommonListParams = {
+  @doc("Number of objects to return per page.")
+  @query
+  maxpagesize?: int32;
+};
+
+model ListJobQueryParams {
+  ...CommonListParams;
+
+  @doc("If specified, filter jobs by status.")
+  @query
+  status?: RouterJobStatusSelector = RouterJobStatusSelector.all;
+
+  @doc("If specified, filter jobs by queue.")
+  @query
+  queueId?: string;
+
+  @doc("If specified, filter jobs by channel.")
+  @query
+  channelId?: string;
+
+  @doc("If specified, filter jobs by classificationPolicy.")
+  @query
+  classificationPolicyId?: string;
+
+  @doc("""
+If specified, filter on jobs that was scheduled before or at given timestamp.
+Range: (-Inf, scheduledBefore].
+""")
+  @query
+  scheduledBefore?: utcDateTime;
+
+  @doc("""
+If specified, filter on jobs that was scheduled at or after given value. Range:
+[scheduledAfter, +Inf).
+""")
+  @query
+  scheduledAfter?: utcDateTime;
+}
+
+model ListClassificationPoliciesQueryParams {
+  ...CommonListParams;
+}
+
+model ListDistributionPoliciesQueryParams {
+  ...CommonListParams;
+}
+
+model ListExceptionPoliciesQueryParams {
+  ...CommonListParams;
+}
+
+model ListQueuesQueryParams {
+  ...CommonListParams;
+}
+
+model ListWorkerQueryParams {
+  ...CommonListParams;
+
+  @doc("If specified, select workers by worker state.")
+  @query
+  state?: RouterWorkerStateSelector = RouterWorkerStateSelector.all;
+
+  @doc("If specified, select workers who have a channel configuration with this channel.")
+  @query
+  channelId?: string;
+
+  @doc("If specified, select workers who are assigned to this queue.")
+  @query
+  queueId?: string;
+
+  @doc("""
+If set to true, select only workers who have capacity for the channel specified
+by `channelId` or for any channel if `channelId` not specified. If set to
+false, then will return all workers including workers without any capacity for
+jobs. Defaults to false.
+""")
+  @query
+  hasCapacity?: boolean;
+}
+
+@doc("Base class for all pageable entities")
+model RouterPageableEntity {
+  @doc("Concurrency Token.")
+  @visibility("read")
+  etag: eTag;
+}
+
+@pagedResult
+@friendlyName("Paged{name}", TResource)
+@doc("Paged collection of {name} items", TResource)
+model RouterPagedCollection<TResource extends TypeSpec.Reflection.Model & RouterPageableEntity> {
+  @doc("The {name} items on this page", TResource)
+  @items
+  value: TResource[];
+
+  @doc("The link to the next page of items")
+  @nextLink
+  nextLink?: string;
+}
```

Comment:
Why not use the paged response in Azure Core?

Reply:
Prior to our SDK review, our `GET`, `PATCH` ops and `LIST` op used to return 2 different objects. E.g., if `GET`, `PATCH` deals with `DistributionPolicy`, `LIST` would return collection of `DistributionPolicyItem` (this was based on our previous rest reviews we did during public preview). Hence, we couldn't use the paged response from Azure Core since LIST wasn't dealing with the same resource. However, we have changed it. Now `GET`, `PATCH` and `LIST` deals with the same entities. I have rectified it now.